views/swap/swap-flip-token.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button } from '@interest-protocol/ui-kit';
+import { useRouter } from 'next/router';
+import { FC } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+
+import { SwapSVG } from '@/svg';
+import { updateURL } from '@/utils';
+
+import { SwapForm } from './swap.types';
+
+const SwapFlipToken: FC = () => {
+  const { pathname } = useRouter();
+  const form = useFormContext<SwapForm>();
+
+  const { setValue, control } = form;
+
+  const to = useWatch({ control, name: 'to' });
+  const from = useWatch({ control, name: 'from' });
+
+  const flipToken = () => {
+    const tmpTo = to;
+    const tmpFrom = from;
+    setValue('to', { ...tmpFrom, value: '' });
+    setValue('from', { ...tmpTo, value: '' });
+
+    updateURL(`${pathname}?from=${tmpTo.type}&to=${tmpFrom.type}`);
+  };
+
+  return (
+    <Button
```

Comment:
This button has a wierd behavior when clicked, the bg change to the same color as the icon. Only change to default bg in next click event.